src/org/intellij/erlang/quickfixes/ErlangCreateFunctionQuickFix.java

```diff
@@ -23,82 +24,105 @@
 import com.intellij.codeInspection.ProblemDescriptor;
 import com.intellij.openapi.editor.Editor;
 import com.intellij.openapi.project.Project;
-import com.intellij.psi.PsiElement;
-import com.intellij.psi.util.PsiTreeUtil;
-import com.intellij.psi.util.PsiUtilBase;
-import com.intellij.util.Function;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.IncorrectOperationException;
 import com.intellij.util.containers.ContainerUtil;
-import org.intellij.erlang.psi.*;
-import org.intellij.erlang.refactoring.ErlangRefactoringUtil;
+import org.jetbrains.annotations.Nls;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.ArrayList;
 import java.util.List;
 
-public class ErlangCreateFunctionQuickFix extends LocalQuickFixBase {
-  private final String myName;
-  private final int myArity;
+public class ErlangCreateFunctionQuickFix extends LocalQuickFixBase implements IntentionAction {
+  private final FunctionTextProvider myFunctionText;
+  private final String myFixMessage;
```

Comment:
You don't need this field - a reference to this object is already stored in superclass.